widgets/contact/contact.php

```diff
@@ -813,36 +813,38 @@ function contact_form_action( $instance, $storage_hash ){
 					continue;
 				}
 			}
+			
+			// Don't process an unrequired empty field
+			if( empty( $value ) ) {
+				continue;
+			}
 
+			// Type Validation
 			switch( $field['type'] ) {
 				case 'email':
-					if( $value != sanitize_email($value) ) {
-						$errors[$field_name] = __('Invalid email address.', 'so-widgets-bundle');
+					if( $value != sanitize_email( $value ) ) {
+						$errors[ $field_name ] = __( 'Invalid email address.', 'so-widgets-bundle' );
 					}
+
```

Comment:
Great idea to tidy up this `switch`, but I'm not really a fan of this style of allowing a `case` to fall through after the `case` is partially handled. It's not clear whether the `break` was left out accidentally or intentionally and it could have unintended effects. I would either repeat the line `$email_fields[ $field['type'] ] = $value;` in the `'email'` case or group the three cases together and have an additional check for the `'email'` type. I.e.
```
	case 'email':
	case 'name':
	case 'subject':
		if( $field['type'] === 'email' && $value != sanitize_email($value) ) {
			$errors[$field_name] = __('Invalid email address.', 'so-widgets-bundle');
		}
		$email_fields[$field['type']] = $value;
		break;
```